baseband/mark5b/base.py

```diff
@@ -338,20 +338,20 @@ class Mark5BStreamWriter(VLBIStreamWriterBase, Mark5BFileWriter):
     """
 
     _frame_class = Mark5BFrame
+    _sample_shape_maker = Mark5BPayload._sample_shape_maker
 
     def __init__(self, raw, sample_rate, nchan=1, bps=2, header=None,
                  squeeze=True, **kwargs):
         if header is None:
             header = Mark5BHeader.fromvalues(**kwargs)
-        sample_shape = Mark5BPayload._sample_shape_maker(nchan)
         super(Mark5BStreamWriter, self).__init__(
-            raw, header0=header, sample_shape=sample_shape, bps=bps,
-            complex_data=False, thread_ids=None,
+            raw, header0=header, unsliced_shape=(nchan,), bps=bps,
+            complex_data=False, subset=None,
             samples_per_frame=header.payloadsize * 8 // bps // nchan,
             sample_rate=sample_rate, squeeze=squeeze)
 
         self._data = np.zeros((self.samples_per_frame,
-                               self._sample_shape.nchan), np.float32)
+                               self._unsliced_shape.nchan), np.float32)
```

Comment:
I spent probably an hour yesterday experimenting with which I liked better.  I like this because it sets a precedent for formats which don't record the sample shape.

Reply:
OK, let's keep as is then.